### PR TITLE
travis: Use a more explicit stages syntax

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
It seems to work and makes more sense to me. I think we should spread this to all projects.

**Comment by leandro-lucarella-sociomantic** [comment]
Maybe first we should check if packages are still being generated properly.

**Comment by mihails-strasuns-sociomantic** [comment]
Yes, would be nice to first check how do stages work when package upload step is taken. I don't fully understand how it interacts with matrix right now.

**Comment by leandro-lucarella-sociomantic** [comment]
I'll test it in my fork.

**Comment by leandro-lucarella-sociomantic** [comment]
It seems to look fine: https://travis-ci.org/leandro-lucarella-sociomantic/dhtnode/builds/325014624 (it will fail uploading because I didn't configure the credentials for jfrog).

**Comment by leandro-lucarella-sociomantic** [comment]
Wait, no, there is a duplicate cell for `DMD=dmd1 F=production`!

**Comment by leandro-lucarella-sociomantic** [comment]
Also for this PR.

**Comment by leandro-lucarella-sociomantic** [comment]
Travis seems to be deeply broken. I can't find a way to make it work. I'm contacting support...

**Comment by leandro-lucarella-sociomantic** [comment]
OK, I got it now. Matrix expansion is an anti-feature. I think it might make sense when you use travis infra to test with a bunch of different versions of the compiler or something like that. But when declaring matrixes, jobs are implicitly created in the "test" stage and there is no way around it. But what we can do is to define the whole matrix *manually* by defining jobs + yaml magic to avoid code duplication, closer to what I saw for circleci in turtle.

I think this makes things more explicit and clear. Unblocking!

**Comment by leandro-lucarella-sociomantic** [comment]
Here are the results with the upload stage (failing because is my fork and it doesn't have credentials): https://travis-ci.org/leandro-lucarella-sociomantic/dhtnode/builds/325124376

File: .travis.yml [hunk 1]
```diff
@@ -11,35 +11,42 @@ services:
...
-# https://github.com/sociomantic-tsunami/beaver/issues/34
-script: beaver dlang make && beaver dlang make pkg
+# Create docker images and setup environment to build
+install: beaver dlang install
 
+# Basic config is inherited from the global scope
 jobs:
     include:
+        # Test matrix
+        - &test-matrix
```
Comment (mihails-strasuns-sociomantic) [inline .travis.yml]: I think it is much cleaner to define templates for anchors in separate yml sections, not used by Travis directly. See https://github.com/sociomantic-tsunami/turtle/blob/v8.x.x/.circleci/config.yml#L16 for example. That way it is more obvious what is different between each matrix row and where to look for all templates if there are many.

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: Yeah, good idea. Updated!

File: .travis.yml [hunk 2]
```diff
@@ -11,35 +11,44 @@ services:
...
+          stage: Test
+          script:
+              - beaver dlang make
+              # Build packages too, this should be done by beaver:
+              # https://github.com/sociomantic-tsunami/beaver/issues/34
+              - beaver dlang make pkg
     include:
+        # Test matrix
+        - <<: *test-matrix
+          env: DMD=dmd1 F=production
```
Comment (mihails-strasuns-sociomantic) [inline .travis.yml]: One last request - while we are touching this, maybe you can switch dhtnode to use pinned compiler versions like we do in ocean/swarm/turtle?

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: Sure, but I will do it in another PR.

**Comment by leandro-lucarella-sociomantic** [comment]
https://github.com/sociomantic-tsunami/dhtnode/pull/61

**Pull request merged**